src/snowflake/snowpark/_internal/analyzer/select_statement.py

```diff
@@ -93,13 +94,14 @@ def __init__(
             Union[str, Expression]
         ] = None,  # used to infer dependent columns
         dependent_columns: Optional[
-            Set[str]
+            AbstractSet[str]
         ] = COLUMN_DEPENDENCY_ALL,  # columns that this column has a dependency on.
         depend_on_same_level: bool = False,  # Whether this column has dependency on one or more columns of the same level instead of the subquery.
         referenced_by_same_level_columns: Optional[
-            Set[str]
+            AbstractSet[str]
         ] = COLUMN_DEPENDENCY_EMPTY,  # Other same-level columns that use this column.
-        state_dict: Optional["ColumnStateDict"] = None,  # has states of all columns.
+        *,
```

Comment:
There's a similar story for `format_type_options` in `SnowflakePlanBuilder.copy_into_table`. In that case I opted for a dirtier fix and a TODO since I was starting to question things, which I'm glad about now lol

Reply:
Hey we don't need to avoid a behavior change here, I'm just curious on why do we need such a change and you explained it pretty well :)